src/license/license.hpp

```diff
@@ -12,26 +12,57 @@
 #pragma once
 
 #include <cstdint>
+#include <optional>
 #include <string>
 
 #include "utils/result.hpp"
 #include "utils/scheduler.hpp"
 #include "utils/settings.hpp"
+#include "utils/spin_lock.hpp"
+#include "utils/synchronized.hpp"
 
-namespace memgraph::utils::license {
+namespace memgraph::license {
+
+enum class LicenseType : uint8_t { ENTERPRISE, OEM };
```

Comment:
Maybe also pass a `LicenseType` parameter to `EnableTesting` to be able specify different license types for tests. It can be defaulted to `LicenseType::ENTERPRISE` to make our life easier.